internal/system/agent/services.go

```diff
@@ -25,35 +29,349 @@ const (
+		json.Unmarshal([]byte(responseJSON), &crp.Config)
+	}
+	return crp
+}
+
+func getConfig(services []string) (ConfigRespMap, error) {
+
+	const (
+		configApiPath = "/api/v1/config"
+	)
+
+	var result = ConfigRespMap{}
 
 	// Loop through requested actions, along with respectively-supplied parameters.
 	for _, service := range services {
-		// TODO: Logging is placeholder for actual functionality to be implemented.
-		LoggingClient.Info(fmt.Sprintf("CONFIG >> Fetching the configuration for the service {%v}!\n", service))
+
+		switch service {
+
+		case edgexSupportNotifications:
+
+			var url string
+
+			if Manifest != nil {
+				url = "http://" + Manifest.EdgexServicesNotificationsHost + ":" + strconv.Itoa(Manifest.EdgexServicesNotificationsPort) + configApiPath
+			}
+			LoggingClient.Debug(fmt.Sprintf("About to fetch the configuration for the service {%v}!", service))
+
+			LoggingClient.Debug("Issuing GET command to: " + url)
+			req, err := http.NewRequest(http.MethodGet, url, nil)
+			if err != nil {
+				LoggingClient.Info("Issuing GET command to: " + url + "failed!")
+				break
+			}
+			resp, err = makeRequest(req)
+			if err != nil {
+				LoggingClient.Info("Issuing GET command to: " + url + "failed!")
+				break
+			}
+			result = processResponse(resp)
+			jsonBytes, err := ConfigRespMap.MarshalJSON(result)
+			LoggingClient.Debug(fmt.Sprintf("For the {%v} service, fetched this configuration: {%v}: ", service, string(jsonBytes)))
+			break
+
+		case edgexCoreCommand:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexCoreCommand))
+			break
+
+		case edgexCoreData:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexCoreData))
+			break
+
+		case edgexCoreMetadata:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexCoreMetadata))
+			break
+
+		case edgexExportClient:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexExportClient))
+			break
+
+		case edgexExportDistro:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexExportDistro))
+			break
+
+		case edgexSupportLogging:
+			LoggingClient.Info(fmt.Sprintf("The micro-service {%v} currently does not support an endpoint for providing its configuration data...", edgexSupportLogging))
+			break
+
+		default:
+			LoggingClient.Warn(fmt.Sprintf(">> Unknown service: %v", service))
+			break
+		}
 	}
-	return true
+	return result, nil
 }
 
-func getMetric(services []string, metrics []string) bool {
+func getMetrics(services []string) (MetricsRespMap, error) {
+
+	const (
+		metricsApiPath = "/api/v1/metrics"
+	)
```

Comment:
Put this elsewhere -- either in internal/constants.go or in /pkg/clients/constants.go

Reply:
Done.